### PR TITLE
Fix issue - semantic props overwritten when single prop used; update semantic to 1.3.1

### PR DESCRIPTION
**Pull request by jacqueswho**
### Reasons for making this change

- Semantic updated to 1.3.1
- getSemanticProps to allow user pass semantic ui props including defaults
- getSemanticErrorProps to allow user to pass ui error props
- fixes this [issue](https://github.com/rjsf-team/react-jsonschema-form/issues/2619)

### Checklist

* [ ] **I'm updating documentation**
  - [X] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [X] **I'm adding or updating code**
  - [X] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [X] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,10 @@ should change the heading of the (upcoming) version to include a major version b
 ## @rjsf/fluent-ui
 - fluent-ui: Allow value of 0 in TextWidget (https://github.com/rjsf-team/react-jsonschema-form/pull/2497)
 
+## @rjsf/semantic-ui
+- "semantic-ui-react" updated to v1.2.1 (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)
+- fixed passing semantic theme specific props, by overiding defaults, with your own semantic props (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)
```
[the remark this answers is not in the document]
Comment (jacqueswho) [reply CHANGELOG.md]: @epicfaace is there anything else

**Comment by epicfaace** [comment]
Sorry for the delay! I will take a look soon.

--
Ashwin Ramaswami

File: packages/semantic-ui/src/util.js [hunk 2]
```diff
@@ -4,21 +4,69 @@ import React from "react";
...
 export function getSemanticProps({
   formContext = {},
   uiSchema = {},
   options = {},
+  defaultSchemaProps = { fluid: true ,inverted: false },
+  defaultContextProps= { }
 }) {
+
+   const hasFormContextProps = formContext.semantic  ? true : false;
+   const hasSchemaProps = uiSchema["ui:options"] && uiSchema["ui:options"].semantic ? true : false;
```
Comment (epicfaace) [inline packages/semantic-ui/src/util.js:23]: I think we should call this `hasUiSchemaProps` instead?

File: packages/semantic-ui/src/util.js [hunk 3]
```diff
@@ -4,21 +4,69 @@ import React from "react";
...
+   const hasOptionsProps = options.semantic  ? true : false;
+   const formContextProps = hasFormContextProps ? formContext.semantic : defaultContextProps;
+   let schemaProps = hasSchemaProps ? uiSchema["ui:options"].semantic : defaultSchemaProps;
+   let optionProps = hasOptionsProps ? options.semantic : {};
+   // formContext props should overide other props
+   if (hasFormContextProps){
+     Object.keys(formContext.semantic).map((key) => {
```
Comment (epicfaace) [inline packages/semantic-ui/src/util.js:30]: Do we need this code to get the formContext prop to override other props? Can't we remove this and afterwards just do:

```
  return Object.assign(
    {},
    schemaProps,
    optionProps,
    formContextProps,
  );
```

File: packages/semantic-ui/src/TextWidget/index.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
-/* eslint-disable import/export */
-export { default } from './TextWidget';
-export * from './TextWidget';
+/* eslint-disable import/export */
+export { default } from './TextWidget';
+export * from './TextWidget';
```
Comment (heath-freenome) [inline packages/semantic-ui/src/TextWidget/index.js:3]: Ditto `index.js`

File: packages/semantic-ui/src/TextareaWidget/TextareaWidget.js [hunk 5]
```diff
@@ -24,7 +23,7 @@ function TextareaWidget(props) {
     uiSchema,
     formContext,
   } = props;
-  const semanticProps = getSemanticProps({ formContext, options });
+  const semanticProps = getSemanticProps({ formContext, options, defaultSchemaProps: {} });
```
Comment (heath-freenome) [inline packages/semantic-ui/src/TextareaWidget/TextareaWidget.js:26]: Given the example from `EmailWidget` should `defaultSchemaProps` be assigned as the following since the defaultProps had that?:
```
    defaultSchemaProps:{
      inverted: false,
    }
```

File: packages/semantic-ui/src/TextareaWidget/index.js [hunk 6]
```diff
@@ -1,3 +1,3 @@
-/* eslint-disable import/export */
-export { default } from "./TextareaWidget";
-export * from "./TextareaWidget";
+/* eslint-disable import/export */
+export { default } from "./TextareaWidget";
+export * from "./TextareaWidget";
```
Comment (heath-freenome) [inline packages/semantic-ui/src/TextareaWidget/index.js:3]: Ditto `index.js`

File: packages/semantic-ui/src/UpDownWidget/index.js [hunk 7]
```diff
@@ -1,3 +1,3 @@
-/* eslint-disable import/export */
-export { default } from './UpDownWidget';
-export * from './UpDownWidget';
+/* eslint-disable import/export */
+export { default } from './UpDownWidget';
+export * from './UpDownWidget';
```
Comment (heath-freenome) [inline packages/semantic-ui/src/UpDownWidget/index.js:3]: Ditto `index.js`

File: packages/semantic-ui/src/Widgets/index.js [hunk 8]
```diff
@@ -1,3 +1,3 @@
-/* eslint-disable import/export */
-export { default } from './Widgets';
-export * from './Widgets';
+/* eslint-disable import/export */
+export { default } from './Widgets';
+export * from './Widgets';
```
Comment (heath-freenome) [inline packages/semantic-ui/src/Widgets/index.js:3]: Ditto `index.js`

File: packages/semantic-ui/src/util.js [hunk 9]
```diff
@@ -4,21 +4,63 @@ import React from "react";
...
+ * @param {Object?} params.options
+ * @param {Object?} params.defaultSchemaProps
+ * @param {Object?} params.defaultContextProps
  * @returns {any}
  */
 export function getSemanticProps({
   formContext = {},
   uiSchema = {},
   options = {},
+  defaultSchemaProps = { fluid: true ,inverted: false },
```
Comment (heath-freenome) [inline packages/semantic-ui/src/util.js:18]: Fix the comma placement... I.e. `{ fluid: true, inverted: false }`

File: packages/semantic-ui/package.json [hunk 10]
```diff
@@ -31,7 +31,7 @@
     "nanoid": "^3.1.23",
     "react": ">=16",
     "semantic-ui-css": "^2.4.1",
-    "semantic-ui-react": "^0.87.3"
+    "semantic-ui-react": "1.2.1"
```
Comment (heath-freenome) [inline packages/semantic-ui/package.json:34]: Why not `v1.3.1` which is the latest of the version 1 `semantic-ui` versions?

File: packages/semantic-ui/src/PasswordWidget/PasswordWidget.js [hunk 11]
```diff
@@ -23,7 +22,16 @@ function PasswordWidget(props) {
...
-  const semanticProps = getSemanticProps({ formContext, options });
+  const semanticProps = getSemanticProps({
+    schema,
+    uiSchema,
+    formContext,
+    options,
+    defaultSchemaProps: {
+      inverted: false,
+      fluid: true,
+    }
```
Comment (heath-freenome) [inline packages/semantic-ui/src/PasswordWidget/PasswordWidget.js:33]: You could also NOT pass in `defaultSchemaProps` since those two values are the default when not present

File: packages/semantic-ui/src/SelectWidget/SelectWidget.js [hunk 12]
```diff
@@ -78,7 +79,18 @@ function SelectWidget(props) {
...
     onFocus,
   } = props;
-  const semanticProps = getSemanticProps({ options });
+  const semanticProps = getSemanticProps({
+    schema,
+    uiSchema,
+    formContext,
+    options,
+    defaultSchemaProps:{
+      inverted: "false",
```
Comment (heath-freenome) [inline packages/semantic-ui/src/SelectWidget/SelectWidget.js:88]: Is making the `inverted` value a string (vs it being a boolean everywhere else) necessary? Consider switching it and seeing if things still work

File: packages/semantic-ui/src/TextWidget/TextWidget.js [hunk 13]
```diff
@@ -1,48 +1,57 @@
...
   const _onFocus = () => onFocus && onFocus(id, value);
-  const inputType = schema.type === 'string' ?  'text' : `${schema.type}`;
+
   const displayLabel = getDisplayLabel(
     schema,
     uiSchema
     /* TODO: , rootSchema */
   );
+
+  const inputType = schema.type === 'string' ?  'text' : `${schema.type}`;
```
Comment (heath-freenome) [inline packages/semantic-ui/src/TextWidget/TextWidget.js:46]: You could just revert these changes since the location of `inputType` in the code does not matter

File: packages/semantic-ui/src/TextWidget/TextWidget.js [hunk 14]
```diff
@@ -1,48 +1,57 @@
...
     formContext,
   } = props;
-  const semanticProps = getSemanticProps({ formContext, options });
+  const semanticProps = getSemanticProps(
+    { formContext, options,
+      uiSchema,
+      defaultSchemaProps: {
+        fluid: true,
+        inverted: false,
+    }
```
Comment (heath-freenome) [inline packages/semantic-ui/src/TextWidget/TextWidget.js:32]: Consider moving `options` to a separate line to be consistent with most of the rest of the code. Also, consider not passing `defaultSchemaProps` since it has the same props set as the default value when it is not provided.

File: packages/semantic-ui/src/URLWidget/URLWidget.js [hunk 15]
```diff
@@ -22,7 +22,14 @@ function URLWidget(props) {
     uiSchema,
     formContext,
   } = props;
-  const semanticProps = getSemanticProps({ formContext, options });
+  const semanticProps = getSemanticProps(
+    { formContext, options,
+      uiSchema,
+      defaultSchemaProps: {
+        fluid: true,
+        inverted: false,
```
Comment (heath-freenome) [inline packages/semantic-ui/src/URLWidget/URLWidget.js:30]: Consider moving `options` to a separate line to be consistent with most of the rest of the code. Also, consider not passing `defaultSchemaProps` since it has the same props set as the default value when it is not provided.

File: packages/semantic-ui/src/UpDownWidget/UpDownWidget.js [hunk 16]
```diff
@@ -9,22 +8,29 @@ const { getDisplayLabel } = utils;
...
     uiSchema,
+    formContext,
   } = props;
-  const semanticProps = getSemanticProps({ formContext, options });
+  const semanticProps = getSemanticProps(
+    { formContext, options,
+      uiSchema,
+      defaultSchemaProps: {
+        fluid: true,
+        inverted: false,
```
Comment (heath-freenome) [inline packages/semantic-ui/src/UpDownWidget/UpDownWidget.js:31]: Consider moving `options` to a separate line to be consistent with most of the rest of the code. Also, consider not passing `defaultSchemaProps` since it has the same props set as the default value when it is not provided.

File: packages/semantic-ui/test/.eslintrc [hunk 17]
```diff
@@ -1,5 +1,5 @@
-{
-  "env": {
-    "jest": true,
-  }
-}
+{
+  "env": {
+    "jest": true,
+  }
+}
```
Comment (heath-freenome) [inline packages/semantic-ui/test/.eslintrc:5]: Same comment as `index.js` about revert due to non-visible End-of-line change

**Review by heath-freenome — CHANGES_REQUESTED** [review]

File: packages/semantic-ui/src/RadioWidget/RadioWidget.js [hunk 18]
```diff
@@ -1,29 +1,35 @@
...
+    { formContext,
+      options,
+      uiSchema,
+      defaultSchemaProps: {}
+    });
   // eslint-disable-next-line no-shadow
-  const _onChange = (event, { value }) =>
-    onChange && onChange(schema.type === "boolean" ? value !== "false" : value);
+  const _onChange = (event, { value : eventValue }) =>{
+    return onChange &&  onChange(schema.type === "boolean" ? eventValue !== "false" : eventValue);};
```
[the remark this answers is not in the document]
Comment (jacqueswho) [reply packages/semantic-ui/src/RadioWidget/RadioWidget.js:31]: I took inspiration from your the bootstrap 4 theme, and I see all themes do the exact same thing.

File: packages/semantic-ui/src/RadioWidget/RadioWidget.js [hunk 19]
```diff
@@ -43,7 +49,7 @@ function RadioWidget({
             label={`${option.label}`}
             value={`${option.value}`}
             key={`${option.value}-${i}`}
-            checked={value === option.value}
+            checked={value == option.value}
```
[the remark this answers is not in the document]
Comment (jacqueswho) [reply packages/semantic-ui/src/RadioWidget/RadioWidget.js:51]: same as mentioned above

File: packages/semantic-ui/src/SelectWidget/SelectWidget.js (hunk 12 above)
Comment (jacqueswho) [reply packages/semantic-ui/src/SelectWidget/SelectWidget.js:88]: the react props warn for the select widget it not being a string. strange I know

**Comment by jacqueswho** [comment]
@heath-freenome thanks for the feedback. made the changes

File: CHANGELOG.md [hunk 20]
```diff
@@ -24,6 +31,7 @@ should change the heading of the (upcoming) version to include a major version b
 - Don't crash when non-object formData is passed in to a schema item with additionalProperties (https://github.com/rjsf-team/react-jsonschema-form/pull/2595)
 - Upgrade jsonpointer to 5.0.0 to address security vulnerability (https://github.com/rjsf-team/react-jsonschema-form/pull/2599)
 
+
```
Comment (heath-freenome) [inline CHANGELOG.md:34]: Consider removing this blank space that was added unnecessarily

File: packages/semantic-ui/src/RadioWidget/RadioWidget.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (heath-freenome) [reply packages/semantic-ui/src/RadioWidget/RadioWidget.js:31]: I actually made the suggested change in my Mui 5 PR so I know it works... But it is up to you

**Review by heath-freenome — APPROVED** [review]
@epicfaace Looks good to me

File: CHANGELOG.md [hunk 21]
```diff
@@ -15,6 +15,13 @@ it according to semantic versioning. For example, if your PR adds a breaking cha
 should change the heading of the (upcoming) version to include a major version bump.
 
 -->
+# v3.3.1 (upcoming)
+
+# v3.3.0
+
+## @rjsf/semantic-ui
+- "semantic-ui-react" updated to v1.3.1 (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)
```
Comment (epicfaace) [inline CHANGELOG.md:23]: Thanks for the review @heath-freenome ! I think the only thing remaining is that I'd like to have a clearer description in the changelog to describe what issue exactly was fixed (rather than just linking to this PR).

Comment (jacqueswho) [reply CHANGELOG.md]: @epicfaace  Im not sure what else you want me to describe,

Comment (epicfaace) [reply CHANGELOG.md]: The changelog right now says "fixed passing [semantic theme specific props](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/themes/semantic-ui/uiSchema/), by only overiding defaults that match, with your own semantic props (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)". That's confusing for a user who is seeing this changelog, and I'm not quite sure what it means either. Maybe we can add more sentences / an example to describe what it's about? I remember you said earlier that "The problem was that if you passed one specific theme prop of your own all other default props were lost."

This is actually why making an issue on GitHub Issues and linking it to this PR will make this much easier. It will then be really clear what exactly the issue is, and we can easily then convert the issue to a changelog description item. Do you think you'd be able to do that?

Comment (jacqueswho) [reply CHANGELOG.md]: @epicfaace thanks, I will create an issue and link it to this PR. But after this is merged in I think we should move this into core util to obtain theme specific props. Question though, should the default theme specific props be over ridden if a user passes his own. for example  these are the default props for a selectWidget 
```JS 
{
      inverted: "false",
      selection: true,
      fluid: true,
      scrolling: true,
      upward: false,
    }
```
should a user now pass 
```JS
{
  'ui:options' : {
     semantic: {
       fluid: false,
     }
   }
}
```

that select dropdown is going to look terrible because it has the lost the rest of the default props.

so should the new props be

```JS
{
      inverted: "false",
      selection: true,
      fluid: false,
      scrolling: true,
      upward: false,
    }
```
OR 
```JS
{
      fluid: false,
 }
```

So leave the default props in place unless that prop is passed exclusively by the user

File: packages/semantic-ui/package.json [hunk 22]
```diff
@@ -31,7 +31,7 @@
     "nanoid": "^3.1.23",
     "react": ">=16",
     "semantic-ui-css": "^2.4.1",
-    "semantic-ui-react": "^0.87.3"
+    "semantic-ui-react": "1.3.1"
```
Comment (epicfaace) [inline packages/semantic-ui/package.json:34]: Are there any breaking changes in this version bump that would make it different to use @rjsf/semantic-ui? If so, we might need to bump the rjsf version to a major version as well.

Comment (epicfaace) [reply packages/semantic-ui/package.json:34]: Thanks for making the issue @jacqueswho! I think this is the only remaining outstanding issue before we can get this merged.

Comment (jacqueswho) [reply packages/semantic-ui/package.json:34]: @epicfaace according to [this ](https://github.com/Semantic-Org/Semantic-UI-React/releases) no.

Comment (jacqueswho) [reply packages/semantic-ui/package.json:34]: @epicfaace I have updated the getSemanticProps function, to make it more simpler

File: packages/semantic-ui/src/util.js [hunk 23]
```diff
@@ -4,21 +4,61 @@ import React from "react";
...
+  defaultContextProps= { }
 }) {
+
+   const formContextProps = formContext.semantic;
+   let schemaProps = uiSchema["ui:options"] && uiSchema["ui:options"].semantic;
+   let optionProps = options.semantic;
+   // formContext props should overide other props
+   return Object.assign(
+    {},
+    { ...(defaultSchemaProps && defaultSchemaProps) },
```
Comment (epicfaace) [inline packages/semantic-ui/src/util.js:28]: What does `defaultSchemaProps && defaultSchemaProps` do? Why not just do `defaultSchemaProps`?

Comment (jacqueswho) [reply packages/semantic-ui/src/util.js:28]: if it is undefined it wont destructure the props. Its quite a neat approach for null/ undefined checks

**Review by epicfaace — APPROVED** [review]
Thanks! Sorry it took so long, the issue helped to clarify what's it about.

File: CHANGELOG.md [hunk 24]
```diff
@@ -15,6 +15,13 @@ it according to semantic versioning. For example, if your PR adds a breaking cha
 should change the heading of the (upcoming) version to include a major version bump.
 
 -->
+# v3.3.1 (upcoming)
+
+# v3.3.0
+
+## @rjsf/semantic-ui
+- "semantic-ui-react" dependency updated to v1.3.1 (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)
+- fixed passing [semantic theme specific props](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/themes/semantic-ui/uiSchema/), by only overiding defaults that match, with your own semantic props (https://github.com/rjsf-team/react-jsonschema-form/pull/2590) related to this [issue 2619](https://github.com/rjsf-team/react-jsonschema-form/issues/2619)
```
Comment (epicfaace) [inline CHANGELOG.md:24]:
```suggestion
- fixed an issue where all semantic props overwritten when a single [semantic theme-specific prop](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/themes/semantic-ui/uiSchema/) is passed in ([issue 2619](https://github.com/rjsf-team/react-jsonschema-form/issues/2619)) (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)
```

File: packages/semantic-ui/test/util.test.js [hunk 25]
```diff
@@ -0,0 +1,99 @@
+import { getSemanticErrorProps, getSemanticProps } from '../src/util';
+
+
+describe("util js functions", () => {
+
+  describe("getSemanticProps", () => {
+
+      test("defaultSchemaProps should be returned", () => {
```
Comment (epicfaace) [inline packages/semantic-ui/test/util.test.js:8]: Actually, just one more thing. Can you add a snapshot test where we add a single semantic-specific prop to override an input field, then render a `<Form>` element, then the snapshot should ensure that the form is not overridden? This way we test the full end-to-end use case for #2619.

Comment (jacqueswho) [reply packages/semantic-ui/test/util.test.js:8]: @epicfaace test updated
